azurerm/internal/services/iotcentral/validate/iotcentral_app_test.go

```diff
@@ -0,0 +1,131 @@
+package validate
+
+import (
+	"testing"
+)
+
+func TestIotCentralAppName(t *testing.T) {
+	validNames := []string{
+		"a1", "11", "1a", "aa",
+		"1-1", "aaa-aa", "a--a-aa",
+		"a1-1", "a1-a", "1a-1", "1a-a-1-2",
+		"abcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcde",
+		"abcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcde123",
```

Comment:
Could we make this consistent with other validation tests in the provideR?

Reply:
fixed